### PR TITLE
Executive 2022-05-18

### PR DESCRIPTION
**Pull request by LongForWisdom**

### CODE REVIEW

File: governance/votes/Executive vote - May 18, 2022.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+In the event of an [Emergency Shutdown](https://docs.makerdao.com/smart-contract-modules/shutdown), the `DssCure` module ensures a fair distribution to DAI holders by calculating how much debt needs to be reduced where there is pre-minted, unused DAI sitting in the `Vat`. Discounting these pre-minted, out-of-circulation DAI is necessary for the correct calculation of the outstanding DAI---otherwise, the artificial, increased debt resulting from factoring in the pre-minted DAI will result in DAI holders receiving less collateral than they are entitled to.
+
+`DssCure` sits between the `Vat` and `End` so that `End` receives the correct amount of debt.
+
+The following additions and changes will take place if this executive proposal passes:
+- A new `DssCure` module will be added to the Maker Protocol.
```
Comment (gbalabasquer) [inline governance/votes/Executive vote - May 18, 2022.md:50]: You might want to call it `MCD_CURE` referring to its future chainlog key.

File: governance/votes/Executive vote - May 18, 2022.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- The `MCD_END` module will be replaced by a new version that interacts with `DssCure`.
```
Comment (gbalabasquer) [inline governance/votes/Executive vote - May 18, 2022.md:51]: Same here.

File: governance/votes/Executive vote - May 18, 2022.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+- The existing ESM contract will be made to point to the new `MCD_END`.
```
Comment (gbalabasquer) [inline governance/votes/Executive vote - May 18, 2022.md:52]: `MCD_ESM`

File: governance/votes/Executive vote - May 18, 2022.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+### Foundational L2 Work
+
+The [Maker Teleport](https://forum.makerdao.com/t/introducing-maker-wormhole/11550) initiative, [previously known as Maker Wormhole](https://forum.makerdao.com/t/maker-wormhole-new-name-maker-teleport/15115), will be bringing canonical DAI capability onto L2 domains. As part of this development, it is necessary to introduce a new module called `DssCure` and update the existing `End` module to interact with it.
+
+In the event of an [Emergency Shutdown](https://docs.makerdao.com/smart-contract-modules/shutdown), the `DssCure` module ensures a fair distribution to DAI holders by calculating how much debt needs to be reduced where there is pre-minted, unused DAI sitting in the `Vat`. Discounting these pre-minted, out-of-circulation DAI is necessary for the correct calculation of the outstanding DAI---otherwise, the artificial, increased debt resulting from factoring in the pre-minted DAI will result in DAI holders receiving less collateral than they are entitled to.
```
Comment (gbalabasquer) [inline governance/votes/Executive vote - May 18, 2022.md:45]: This explanation is correct but refers specifically to L2 MCD DAI.
When L2 -> L2 teleport or L2 lending is available the `Cure` will be on charge to calculate the `gap` between pre minted DAI and lent DAI and report that back as non circulating DAI, meaning that needs to be removed from `End.debt`.
But `Cure` will be first used for L2 -> L1 Teleport, which will report whatever amount of DAI was withdrawn by users but still not burned by the bridge as it is in the 7 days window. It is not strictly the same explanation.
Maybe we want to add it, or make a bit more general the use of it.

File: governance/votes/Executive vote - May 18, 2022.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+For more details, please read [this forum post by the Protocol Engineering Core Unit (PE-001)](https://forum.makerdao.com/t/wednesday-18th-may-executive-dsscure-technical-enhancement/15175).
+
+### New [Flash Mint Module](https://manual.makerdao.com/module-index/module-flash-mint-module)
+
+As described in this [forum post](https://forum.makerdao.com/t/wednesday-18th-may-executive-flash-mint-module-technical-enhancement/15176) if this executive proposal passes the following changes will occur:
+* A new Flash Mint Module will be deployed and authorized.
+* The new module will have no fee parameter and charge no fee in order to save on gas.
+* The Debt Ceiling will be set to 250M on the new module and reduced to 250M on the existing module to maintain 500M total.
+* Both modules will be available in the chainlog as MCD_FLASH_LEGACY and MCD_FLASH.
```
Comment (gbalabasquer) [inline governance/votes/Executive vote - May 18, 2022.md:62]: Maybe we want to clarify that eventually the old module will be removed and the new one will be the only active.

File: governance/votes/Executive vote - May 18, 2022.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+Unless otherwise noted, the changes and additions listed above are subject to the [GSM Pause Delay](https://manual.makerdao.com/parameter-index/core/param-gsm-pause-delay). This means that if this executive proposal passes, the changes and additions listed above will only become active in the Maker Protocol after the GSM Pause Delay has expired. The GSM Pause Delay is currently set to **48 hours**.
+
+If this executive proposal does not pass within 30 days, then it will expire and can no longer have any effect on the Maker Protocol.
+
+---
+
+## Proposal Details
+
+### Core Unit MKR Vesting Streams
```
Comment (patrick-j-govalpha) [inline governance/votes/Executive vote - May 18, 2022.md:37]: This isn't a stream, distribution might be a better word to use

**Review by patrick-j-govalpha — CHANGES_REQUESTED** [review]

File: governance/votes/Executive vote - May 18, 2022.md (hunk 1 above)
Comment (LongForWisdom) [reply governance/votes/Executive vote - May 18, 2022.md]: Kept DssCure references, but clarified it will be added to the chainlog as MCD_CURE.

**Pull request merged**